src/vm/fieldmarshaler.cpp

```diff
@@ -174,7 +174,8 @@ do                                                      \
             TypeHandle pNestedType = fsig.GetLastTypeHandleThrowing(ClassLoader::LoadTypes,
                                                                     CLASS_LOAD_APPROXPARENTS,
                                                                     TRUE);
-            if (pNestedType.GetMethodTable()->IsManagedSequential())
+            if (pNestedType.GetMethodTable()->IsManagedSequential() &&
+                !pNestedType.GetMethodTable()->IsByRefLike())
```

Comment:
I believe this may disable struct promotion on Span<T> as I believe that ContainsPointers() is false. See Compiler::StructPromotionHelper::CanPromoteStructType in the jit, in particular if (StructHasCustomLayout(typeFlags) && ((typeFlags & CORINFO_FLG_CONTAINS_GC_PTR) == 0))

StructHasCustomLayout is another name for !ManagedSequential